tests/rhtap-demo/rhtap-demo.go

```diff
@@ -167,24 +150,32 @@ var _ = framework.RhtapDemoSuiteDescribe(func() {
 					var componentNewBaseBranch, gitRevision string
 					componentRepositoryName := utils.ExtractGitRepositoryNameFromURL(componentSpec.GitSourceUrl)
 					componentList := []*appservice.Component{}
-					var secret string
+					var secretName string
 
 					if componentSpec.Private {
-						secret = SPIGithubSecretName
-						It(fmt.Sprintf("injects manually SPI token for component %s", componentSpec.Name), Label(devEnvTestLabel, stageEnvTestLabel), func() {
-							// Inject spi tokens to work with private components
-							if componentSpec.ContainerSource != "" {
-								// More info about manual token upload for quay.io here: https://github.com/redhat-appstudio/service-provider-integration-operator/pull/115
-								oauthCredentials := `{"access_token":"` + utils.GetEnv(constants.QUAY_OAUTH_TOKEN_ENV, "") + `", "username":"` + utils.GetEnv(constants.QUAY_OAUTH_USER_ENV, "") + `"}`
-
-								_ = fw.AsKubeAdmin.SPIController.InjectManualSPIToken(namespace, componentSpec.ContainerSource, oauthCredentials, corev1.SecretTypeDockerConfigJson, SPIQuaySecretName)
+						It(fmt.Sprintf("creates a secret for private component %s", componentSpec.Name), Label(devEnvTestLabel, stageEnvTestLabel), func() {
+							privateCompSecret := &corev1.Secret{
+								ObjectMeta: metav1.ObjectMeta{
+									Name: constants.PrivateComponentSecretName,
+									Namespace: namespace,
+									Labels: map[string]string {
+										"appstudio.redhat.com/credentials": "scm",
+										"appstudio.redhat.com/scm.host": "github.com",
+									},
+								},
+								Type: corev1.SecretTypeBasicAuth,
+								StringData: map[string]string {
+									"password": os.Getenv("GITHUB_TOKEN"),
```

Comment:
after looking into details of git-clone task, just found out this additional change is required:
```suggestion
									"username": "git",
									"password": os.Getenv("GITHUB_TOKEN"),
```

Reply:
I'm afraid it should be 2 different secrets than. 1st with `password` only for the client instantiation (can be GitHub app as well), with `scm` and `scm.host` labels.   2nd with the `username` and `password` and mentioned in the secret field of the component.

Reply:
fortunately the single secret works just fine. just validated it on my cluster.

maybe it's not the proper way it should be set up, but we will move away from "simple build pipeline" soon in e2e-tests: https://issues.redhat.com/browse/KONFLUX-3654

Reply:
Thanks for your help guys!